spec/apis/register-application.spec.js

```diff
@@ -0,0 +1,27 @@
+export default function() {
+  describe('registerApplication', function() {
+    let app
+    beforeEach(() => {
+      app = {
+        mount(){
+          return Promise.resolve()
+        },
+        unmount() {
+          return Promise.resolve()
+        },
+        bootstrap() {
+          return Promise.resolve()
+        }
+      }
+    })
+
+    it('should throw when I register the same app twice', () => {
+      singleSpa.registerApplication('duplicateApp', app, () => true)
+      expect(
+        () => {
+          singleSpa.registerApplication('duplicateApp', app, () => true)
+        }
+      ).toThrow()
+    })
+  })
+}
```

Comment:
Test to make sure we don't accidentally break the logic that enforces that applications cannot have the same name again in the future.